### PR TITLE
add GTM script tag 

### PR DESCRIPTION
**Pull request by ChisomOguibe11**
## What?
- Setup google tag manager alongside ga4
- push custom `pageLoad` events to track page visits

## Why?
- Requested by business for ETA 

## How?
- If a GTM tag is present in the service hof-config, the gtm script will run across all pages the form 

## Testing?
- Published beta package to test locally and on branch using GTM preview mode

## Anything Else?

### CODE REVIEW

File: frontend/template-partials/views/partials/gatag.html [hunk 1]
```diff
@@ -59,3 +59,25 @@
     }
   </script>
 {{/ga4TagId}}
+
+{{#gtmTagId}}
+  <!-- Data Layer -->
```
Comment (sulthan-ahmed) [inline frontend/template-partials/views/partials/gatag.html:64]: what do you mean by Data layer? was this comment from you?

File: lib/ga-tag.js [hunk 2]
```diff
@@ -2,6 +2,16 @@
...
+const convertToGTMPage = (text) => {
+  // Remove leading and trailing slashes
+  let str = text.replace(/^\/|\/$/g, '');
+  // Replace hyphens with spaces and capitalize each word
+  str = str.replace(/-+/g, ' ').replace(/(^|\s)\S/g, function(match) {
+    return match.toUpperCase();
+  });
+  return str;
+}
+
```
Comment (sulthan-ahmed) [inline lib/ga-tag.js:14]: Looks like github actions is complaining about some linting and styling

File: lib/ga-tag.js [hunk 3]
```diff
@@ -39,22 +49,28 @@ const setupPageMap = routes => {
   return pageMap;
 };
 
-module.exports = (app, config) => {
+  module.exports = (app, config) => {
   const gaTagId = config.gaTagId;
```
Comment (sulthan-ahmed) [inline lib/ga-tag.js:53]: lots of linting issues

File: lib/ga-tag.js [hunk 4]
```diff
@@ -39,22 +49,28 @@ const setupPageMap = routes => {
...
   if (gaTagId || ga4TagId) {
     const pageMap = setupPageMap(routes);
 
     app.use((req, res, next) => {
+      const page = pageView(req.path, pageMap);
       res.locals.gaAllowDebug = config.env === 'development';
       res.locals.gaTagId = gaTagId;
       res.locals.ga4TagId = ga4TagId;
+      res.locals.gtmTagId = gtmTagId
+      res.locals.appName = appName
```
Comment (sulthan-ahmed) [inline lib/ga-tag.js:69]: why do you need the appName?

**Review by sulthan-ahmed — COMMENTED** [review]
looks good just needs some minor changes

File: frontend/template-partials/views/partials/gatag.html (hunk 1 above)
Comment (ChisomOguibe11) [reply frontend/template-partials/views/partials/gatag.html:64]: it's from the snippet code sent from the business. I think it's just a comment to explain what the code is for.

File: lib/ga-tag.js (hunk 2 above)
Comment (ChisomOguibe11) [reply lib/ga-tag.js:14]: aaah thanks ...god I hate linting 😂

File: lib/ga-tag.js (hunk 4 above)
Comment (ChisomOguibe11) [reply lib/ga-tag.js:69]: I was using it before, but no longer need it. Good spot

**Review by shamiluwais — APPROVED** [review]

**Pull request merged**